Mathlib/NumberTheory/LSeries/HurwitzZetaOdd.lean

```diff
@@ -0,0 +1,592 @@
+  rw [← hasSum_ofReal, oddKernel_def' a x]
+  have h1 := hasSum_jacobiTheta₂_term (a * I * x) (by rwa [I_mul_im, ofReal_re] : 0 < im (I * x))
+  have h2 := hasSum_jacobiTheta₂'_term (a * I * x) (by rwa [I_mul_im, ofReal_re] : 0 < im (I * x))
+  refine (((h2.div_const (2 * π * I)).add (h1.mul_left ↑a)).mul_left
+    (cexp (-π * a ^ 2 * x))).congr_fun (fun n ↦ ?_)
+  rw [jacobiTheta₂'_term, mul_assoc (2 * π * I), mul_div_cancel_left₀ _ two_pi_I_ne_zero, ← add_mul,
+    ← mul_assoc, mul_comm _ (n + a : ℂ), mul_assoc (n + a : ℂ), jacobiTheta₂_term,
+    ← Complex.exp_add]
+  push_cast
+  congr 2
+  simp only [← mul_assoc, ← add_mul]
+  congr 1
+  simp_rw [mul_assoc _ I, mul_comm I _, ← mul_assoc _ _ I, mul_assoc _ I, mul_comm I _,
+    ← mul_assoc _ _ I, add_mul, mul_assoc _ _ I, I_mul_I, mul_neg, mul_one,
+    mul_comm (2 : ℂ) π, mul_assoc, ← neg_mul, ← mul_add]
+  rw [← add_assoc, add_comm, mul_comm (n : ℂ) a, ← mul_assoc, add_sq]
+
+lemma hasSum_int_sinKernel (a : ℝ) {t : ℝ} (ht : 0 < t) : HasSum
+    (fun n : ℤ ↦ -I * n * cexp (2 * π * I * a * n) * rexp (-π * n ^ 2 * t)) ↑(sinKernel a t) := by
+  have h : -2 * (π : ℂ) ≠ (0 : ℂ) := by
+    simp? [pi_ne_zero] says simp only [neg_mul, ne_eq, neg_eq_zero, mul_eq_zero,
+      OfNat.ofNat_ne_zero, ofReal_eq_zero, pi_ne_zero, or_self, not_false_eq_true]
+  rw [sinKernel_def]
+  have := hasSum_jacobiTheta₂'_term a (by rwa [I_mul_im, ofReal_re])
+  refine (this.div_const (-2 * π : ℂ)).congr_fun fun n ↦ ?_
+  rw [jacobiTheta₂'_term, jacobiTheta₂_term, ofReal_exp, mul_assoc (-I * n), ← Complex.exp_add,
+    eq_div_iff h, ofReal_mul, ofReal_mul, ofReal_pow, ofReal_neg, ofReal_intCast,
+    mul_comm _ (-2 * π : ℂ), ← mul_assoc]
+  congrm ?_ * cexp (?_ + ?_)
+  · simp only [neg_mul, mul_neg, neg_neg, mul_assoc]
+  · exact mul_right_comm (2 * π * I) a n
+  · simp only [← mul_assoc, mul_comm _ I, I_mul_I, neg_one_mul]
+
+lemma hasSum_nat_sinKernel (a : ℝ) {t : ℝ} (ht : 0 < t) :
+    HasSum (fun n : ℕ ↦ 2 * n * Real.sin (2 * π * a * n) * rexp (-π * n ^ 2 * t))
+    (sinKernel ↑a t) := by
+  rw [← hasSum_ofReal]
+  have := (hasSum_int_sinKernel a ht).sum_nat_of_sum_int
+  simp only [Int.cast_zero, sq (0 : ℂ), zero_mul, mul_zero, add_zero] at this
+  refine this.congr_fun fun n ↦ ?_
+  simp_rw [Int.cast_neg, neg_sq, mul_neg, ofReal_mul, Int.cast_natCast, ofReal_natCast,
+      ofReal_ofNat, ← add_mul, ofReal_sin, Complex.sin]
+  push_cast
+  congr 1
+  rw [← mul_div_assoc, ← div_mul_eq_mul_div, ← div_mul_eq_mul_div, div_self two_ne_zero, one_mul,
+    neg_mul, neg_mul, neg_neg, mul_comm _ I, ← mul_assoc, mul_comm _ I, neg_mul,
+    ← sub_eq_neg_add, mul_sub]
+  congr 3 <;> ring
+
+end sum_formulas
+
+section asymp
+/-!
+## Asymptotics of the kernels as `t → ∞`
+-/
+
+/-- The function `oddKernel a` has exponential decay at `+∞`, for any `a`. -/
+lemma isBigO_atTop_oddKernel (a : UnitAddCircle) :
+    ∃ p, 0 < p ∧ IsBigO atTop (oddKernel a) (fun x ↦ Real.exp (-p * x)) := by
+  obtain ⟨b, _, rfl⟩ := a.eq_coe_Ico
```

Comment:
`induction' a using QuotientAddGroup.induction_on with b` (like elsewhere) would also work (and is slightly faster). Is there some reason to not be consistent?